batchflow/tests/dataformat_test.py

```diff
@@ -140,41 +139,36 @@ def test_block_dataformat(self, model_and_config):
         to train it on a small batch.
-
     """
-    def test_last_default(self, model, model_setup, model_pipeline):
-        """ Default value for 'data_format' is 'channels_last'. """
-        fake_dataset, config = model_setup(data_format='channels_last')
-        pipeline_config = {'model_class': model, 'model_config': config}
-        test_pipeline = model_pipeline.set_config(pipeline_config)
-        total_pipeline = test_pipeline << fake_dataset
-        batch = total_pipeline.next_batch(7, n_epochs=None)
-        assert len(batch) == 7
-
-    def test_last_common(self, model, model_setup, model_pipeline):
-        """ We can explicitly pass 'data_format', it has no effect in this case. """
-        fake_dataset, config = model_setup(data_format='channels_last')
-        config['common/data_format'] = 'channels_last'
-        pipeline_config = {'model_class': model, 'model_config': config}
-        test_pipeline = model_pipeline.set_config(pipeline_config)
-        total_pipeline = test_pipeline << fake_dataset
-        batch = total_pipeline.next_batch(7, n_epochs=None)
-        assert len(batch) == 7
-
-    @pytest.mark.xfail(run=True)
-    def test_first_common(self, model, model_setup, model_pipeline):
-        """ That is intended way to communicate 'data_format' with model. """
-        fake_dataset, config = model_setup(data_format='channels_first')
-        config['common/data_format'] = 'channels_first'
-        pipeline_config = {'model_class': model, 'model_config': config}
-        test_pipeline = model_pipeline.set_config(pipeline_config)
-        total_pipeline = test_pipeline << fake_dataset
-        batch = total_pipeline.next_batch(7, n_epochs=None)
-        assert len(batch) == 7
+    @pytest.mark.parametrize('location', ['common', 'inputs/images'])
+    @pytest.mark.parametrize('data_format',
+                             [None,
+                              pytest.param('channels_first', marks=pytest.mark.xfail),
+                              'channels_last'])
+    def test_data_format(self, model, model_setup, pipeline, location, data_format):
+        """ We can explicitly pass 'data_format' to inputs or common
+
+        Notes
+        -----
+        If `data_format` is None, use a default value.
+
+        `channels_first` does not work in TF 1.12 on CPU as a corresponding pooling operation is not implemented yet.
+        """
+        expected_data_format = data_format or 'channels_last'
+        dataset, model_config = model_setup(data_format=expected_data_format)
+        if data_format:
+            model_config[location + '/data_format'] = data_format
+        config = {'model_class': model, 'model_config': model_config}
+        test_pipeline = (pipeline << dataset).set_config(config)
+        batch = test_pipeline.next_batch(2, n_epochs=None)
+        model = test_pipeline.get_model_by_name('model')
+
+        assert model.data_format('images') == expected_data_format
```

Comment:
```suggestion
         if location == 'inputs/images':
            assert model.data_format('images') == expected_data_format
```
Otherwise it unnecessarily fails when 'channels_first' is passed through 'common'.